aws/kubernetes/MNMG_XGBoost.ipynb

```diff
@@ -0,0 +1,902 @@
+    "    def __init__(self):\n",
+    "        self.start = None\n",
+    "        self.end = None\n",
+    "        self.elapsed = None\n",
+    "\n",
+    "    def __enter__(self):\n",
+    "        self.start = time.perf_counter_ns()\n",
+    "        return self\n",
+    "\n",
+    "    def __exit__(self, exc_type, exc_val, exc_tb):\n",
+    "        self.end = time.perf_counter_ns()\n",
+    "        self.elapsed = self.end - self.start\n",
+    "\n",
+    "        \n",
+    "def create_pod_from_yaml(yaml_file):\n",
+    "    with open(yaml_file, 'r') as reader:\n",
+    "        d = yaml.safe_load(reader)\n",
+    "        d = dask.config.expand_environment_variables(d)\n",
+    "    return make_pod_from_dict(d)\n",
+    "\n",
+    "\n",
+    "def build_worker_and_scheduler_pods(sched_spec, worker_spec):\n",
+    "    assert os.path.isfile(sched_spec)\n",
+    "    assert os.path.isfile(worker_spec)\n",
+    "\n",
+    "    sched_pod = create_pod_from_yaml(sched_spec)\n",
+    "    worker_pod = create_pod_from_yaml(worker_spec)\n",
+    "\n",
+    "    return sched_pod, worker_pod\n",
+    "\n",
+    "def scale_workers(client, n_workers, timeout=300):\n",
+    "    client.cluster.scale(n_workers)\n",
+    "    \n",
+    "    m = len(client.has_what().keys())    \n",
+    "    start = end = time.perf_counter_ns()\n",
+    "    while ((m != n_workers) and (((end - start) / 1e9) < timeout) ):\n",
+    "        time.sleep(5)\n",
+    "        m = len(client.has_what().keys())\n",
+    "        \n",
+    "        end = time.perf_counter_ns()\n",
+    "        \n",
+    "    if (((end - start) / 1e9) >= timeout):\n",
+    "        raise RuntimeError(f\"Failed to rescale cluster in {timeout} sec.\"\n",
+    "              \"Try increasing timeout for very large containers, and verify available compute resources.\")\n",
+    "        "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "dask.config.set({\"logging.kubernetes\": \"info\",\n",
+    "                 \"logging.distributed\": \"info\",\n",
+    "                 \"kubernetes.scheduler-service-type\": \"LoadBalancer\",\n",
+    "                 \"kubernetes.idle-timeout\": None,\n",
+    "                 \"kubernetes.scheduler-service-wait-timeout\": 3600,\n",
+    "                 \"kubernetes.deploy-mode\": \"remote\",\n",
+    "                 \"kubernetes.logging\": \"info\",\n",
+    "                 \"distributed.logging\": \"info\",\n",
+    "                 \"distributed.scheduler.idle-timeout\": None,\n",
+    "                 \"distributed.scheduler.locks.lease-timeout\": None,\n",
+    "                 \"distributed.comm.timeouts.connect\": 3600,\n",
+    "                 \"distributed.comm.tls.ca-file\": certifi.where()})\n",
+    "\n",
+    "sched_spec_path = \"./podspecs/aws/scheduler-specs_new.yml\"\n",
+    "worker_spec_path = \"./podspecs/aws/cuda-worker-specs_new.yml\""
```

Comment:
Is there a reason these have different names than the ones we have provided?

Getting `AssertionError` on the file check `assert os.path.isfile(sched_spec)` needs user to change the name. Is this on purpose? If so, add a comment for that.

Reply:
Nope, typo. I forgot to revert it back to the default yamls to be used.